plugin/src/main/java/org/opensearch/ml/task/MLPredictTaskRunner.java

```diff
@@ -320,17 +320,26 @@ private void runPredict(
                     if (!predictor.isModelReady()) {
                         throw new IllegalArgumentException("Model not ready: " + modelId);
                     }
-                    MLOutput output = mlModelManager.trackPredictDuration(modelId, () -> predictor.predict(mlInput));
-                    if (output instanceof MLPredictionOutput) {
-                        ((MLPredictionOutput) output).setStatus(MLTaskState.COMPLETED.name());
+                    if (mlInput.getAlgorithm() == FunctionName.REMOTE) {
+                        long startTime = System.nanoTime();
+                        ActionListener<MLTaskResponse> trackPredictDurationListener = ActionListener.wrap(output -> {
+                            handleAsyncMLTaskComplete(mlTask);
+                            mlModelManager.trackPredictDuration(modelId, startTime);
+                            internalListener.onResponse(output);
+                        }, e -> handlePredictFailure(mlTask, internalListener, e, false, modelId));
+                        predictor.asyncPredict(mlInput, trackPredictDurationListener);
+                    } else {
+                        MLOutput output = mlModelManager.trackPredictDuration(modelId, () -> predictor.predict(mlInput));
+                        if (output instanceof MLPredictionOutput) {
+                            ((MLPredictionOutput) output).setStatus(MLTaskState.COMPLETED.name());
+                        }
+                        // Once prediction complete, reduce ML_EXECUTING_TASK_COUNT and update task state
+                        handleAsyncMLTaskComplete(mlTask);
+                        internalListener.onResponse(new MLTaskResponse(output));
                     }
-
-                    // Once prediction complete, reduce ML_EXECUTING_TASK_COUNT and update task state
-                    handleAsyncMLTaskComplete(mlTask);
-                    MLTaskResponse response = MLTaskResponse.builder().output(output).build();
-                    internalListener.onResponse(response);
                     return;
                 } catch (Exception e) {
+                    log.error("Failed to predict model " + modelId, e);
```

Comment:
please fix this issue before 2.14 code freeze

Reply:
@zane-neo I raised a PR for this: https://github.com/opensearch-project/ml-commons/pull/2310

Yeah it might not cover all the logs, but most of the customer facing exception. Let's follow this pattern and apply wherever necessary.